middleware/middleware18_test.go

```diff
@@ -0,0 +1,71 @@
+// +build go1.8
+
+package middleware
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+// NOTE: we must import `golang.org/x/net/http2` in order to explicitly enable
+// http2 transports for certain tests. The runtime pkg does not have this dependency
+// though as the transport configuration happens under the hood on go 1.7+.
+
+func TestWrapWriterHTTP2(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, cn := w.(http.CloseNotifier)
+		if !cn {
+			t.Fatal("request should have been a http.CloseNotifier")
```

Comment:
Now I think about it, these messages should probably all read `ResponseWriter should have been ...`